### PR TITLE
Added the Reproductive History section to the Fanconi Survey

### PR DESCRIPTION
**Pull request by sonika-nih**
Added the Reproductive History section to the Fanconi Survey in Playwright

### CODE REVIEW

File: src/test/java/ServiceNow/CHARMS/Steps/FHQChildRecordProducerSteps.java
Comment (Mariachaudhry) [inline src/test/java/ServiceNow/CHARMS/Steps/FHQChildRecordProducerSteps.java:1]: There are some extra lines in Imports and within methods to remove.

**Review by Mariachaudhry — CHANGES_REQUESTED** [review]

File: src/test/java/ServiceNow/CHARMS/Features/FHQ/FHQSubmission.feature [hunk 1]
```diff
@@ -2,15 +2,15 @@
 Feature: FHQ Form Submission Scenarios
   Description:This feature file contains scenarios which submit FHQ Screener by Participant or Legal guardian of the participant in the portal and assertion of data in the NativeView
 
-  @jains @FHQSubmissionInPortal @CP2-562 @CP2-930 @CP2-559 @Regression @selenium
+  @jains @FHQSubmissionInPortal @CP2-562 @CP2-930 @CP2-559 @Regression1 @selenium
```
Comment (hamidnesar) [inline src/test/java/ServiceNow/CHARMS/Features/FHQ/FHQSubmission.feature:5]: Please modify the tag to @Regression

File: src/test/java/ServiceNow/CHARMS/Features/FHQ/FHQSubmission.feature [hunk 2]
```diff
@@ -64,9 +64,9 @@ Feature: FHQ Form Submission Scenarios
 
     Examples:
       | Email                | Password      | FHQScenario  | FHQRelative |
-      | charmsras2@yahoo.com | RASTest2023$$ | FHQScenario1 | Children    |
+      | charmsras1@yahoo.com | RASTest2022$$ | FHQScenario1 | Children    |
 
-  @jains @FHQSubmissionInPortal @CP2-603 @CP2-1020 @Regression @selenium
+  @jains @FHQSubmissionInPortal @CP2-603 @CP2-1020 @Regression1 @selenium
```
Comment (hamidnesar) [inline src/test/java/ServiceNow/CHARMS/Features/FHQ/FHQSubmission.feature:69]: Modify the tag to @Regression.
Regression vs Regression1

File: src/test/java/ServiceNow/CHARMS/Runners/RunCHARMSFHQRegressionTest.java [hunk 3]
```diff
@@ -7,8 +7,8 @@
 		"rerun:target/failed.txt",
 		"pretty", "com.aventstack.extentreports.cucumber.adapter.ExtentCucumberAdapter:"},
 features = "src/test/java/ServiceNow/CHARMS/Features/FHQ",
-glue = "ServiceNow.CHARMS.Steps", 
-tags = "@Regression", 
+glue = {"ServiceNow.CHARMS.Steps", "Hooks"},
+tags = "@Regression1",
```
Comment (hamidnesar) [inline src/test/java/ServiceNow/CHARMS/Runners/RunCHARMSFHQRegressionTest.java:11]: Regression vs Regression1

File: src/test/java/ServiceNow/CHARMS/Steps/FHQChildRecordProducerSteps.java
Comment (hamidnesar) [reply src/test/java/ServiceNow/CHARMS/Steps/FHQChildRecordProducerSteps.java:1]: 1. Please remove empty lines: 
4, 26, 28, 31, 34, 36, 41, 43, 46, 48, 193, 229, 232, 235, 241, 244, 247, 250, 253, 256, 259, 262, 268, 271, 285, 294, 297, 300, 304, 307, 312, 314, and _MANY MORE_ throughout the class. 

2. Methods on line 51, 95, 118, 160, 174, 184, 203, 213, 227, 289 and _MANY MORE_ throughout the class are missing Javadoc comments.

**Review by hamidnesar — CHANGES_REQUESTED** [review]
Hi, 
Please review the comments. Thank you!

**Review by Mariachaudhry — APPROVED** [review]

File: src/test/java/ServiceNow/CHARMS/Steps/FHQChildRecordProducerSteps.java
Comment (hamidnesar) [inline src/test/java/ServiceNow/CHARMS/Steps/FHQChildRecordProducerSteps.java:1]: Methods on line 32, 40, 84, 107, 149, 163, 173, 191, 201, and _MANY MORE_ throughout the class are missing Javadoc comments.

**Review by hamidnesar — CHANGES_REQUESTED** [review]
Hi, 
Please review the comments. thanks

File: src/test/java/ServiceNow/CHARMS/Steps/FHQChildRecordProducerSteps.java
Comment (sonika-nih) [reply src/test/java/ServiceNow/CHARMS/Steps/FHQChildRecordProducerSteps.java:1]: Done

Comment (sonika-nih) [reply src/test/java/ServiceNow/CHARMS/Steps/FHQChildRecordProducerSteps.java:1]: old code. WIll do the cleaning as time permits

File: src/test/java/ServiceNow/CHARMS/Features/FHQ/FHQSubmission.feature (hunk 1 above)
Comment (sonika-nih) [reply src/test/java/ServiceNow/CHARMS/Features/FHQ/FHQSubmission.feature]: done

File: src/test/java/ServiceNow/CHARMS/Features/FHQ/FHQSubmission.feature (hunk 2 above)
Comment (sonika-nih) [reply src/test/java/ServiceNow/CHARMS/Features/FHQ/FHQSubmission.feature]: Done

File: src/test/java/ServiceNow/CHARMS/Runners/Run_CHARMS_Fanconi_Regression_Test.java
Comment (iamdez99) [inline src/test/java/ServiceNow/CHARMS/Runners/Run_CHARMS_Fanconi_Regression_Test.java:1]: Can you please change this tag to @Fanconi_Regression? And update all fanconi regression scenarios with this tag.

File: src/test/java/ServiceNow/CHARMS/Features/FHQ/FHQSubmission.feature
Comment (iamdez99) [inline src/test/java/ServiceNow/CHARMS/Features/FHQ/FHQSubmission.feature:1]: I think charmsras1@yahoo.com is the account I use for my tests. Will there be any issues if you use it here?

File: src/test/java/ServiceNow/CHARMS/Pages/FHQSurveyPortalPage.java
Comment (iamdez99) [inline src/test/java/ServiceNow/CHARMS/Pages/FHQSurveyPortalPage.java:1]: For the xpath, you are pasting a String within a String, thats why it looks like "//span[normalize-space()=\"Are any of the following individuals a biological parent of this child?\"]", instead it should be "//span[normalize-space()='Are any of the following individuals a biological parent of this child?']"

File: src/test/java/ServiceNow/CHARMS/Runners/RunCHARMSFHQRegressionTest.java
Comment (iamdez99) [inline src/test/java/ServiceNow/CHARMS/Runners/RunCHARMSFHQRegressionTest.java:1]: Can you please change this tag @FHQ_Regression and update all FHQ regression scenarios with this tag.

**Review by iamdez99 — CHANGES_REQUESTED** [review]
Please see comments.

File: src/test/java/ServiceNow/CHARMS/Runners/RunCHARMSFHQRegressionTest.java
Comment (sonika-nih) [reply src/test/java/ServiceNow/CHARMS/Runners/RunCHARMSFHQRegressionTest.java:1]: Done

File: src/test/java/ServiceNow/CHARMS/Features/FHQ/FHQSubmission.feature
Comment (sonika-nih) [reply src/test/java/ServiceNow/CHARMS/Features/FHQ/FHQSubmission.feature:1]: Changed to another test account

File: src/test/java/ServiceNow/CHARMS/Runners/Run_CHARMS_Fanconi_Regression_Test.java
Comment (sonika-nih) [reply src/test/java/ServiceNow/CHARMS/Runners/Run_CHARMS_Fanconi_Regression_Test.java:1]: Done

File: src/test/java/ServiceNow/CHARMS/Pages/FHQSurveyPortalPage.java
Comment (sonika-nih) [reply src/test/java/ServiceNow/CHARMS/Pages/FHQSurveyPortalPage.java:1]: This is another way of writing the xpath. both works. For eg given below are both correct
 (//*[contains(text(),"sub-title")])
(//*[contains(text(),'sub-title')])

**Review by hamidnesar — APPROVED** [review]
Reviewed and approved!

**Review by iamdez99 — APPROVED** [review]

**Pull request merged**